src/main/services/UserSettingService.ts

```diff
@@ -64,16 +67,36 @@ class UserSettingService {
   }
 
   public async read(path: string) {
-    this.setMyPath(path);
-    if (!(await wsUtils.fileExist(`${this.myPath}/${this.name}`)))
-      await this.save();
-    const setting = await fs.promises.readFile(
-      `${this.myPath}/${this.name}`,
-      'utf8'
-    );
-    this.store = JSON.parse(setting);
+    try {
+      this.setMyPath(path);
+      if (!(await wsUtils.fileExist(`${this.myPath}/${this.name}`)))
+        await this.save();
+      const setting = await fs.promises.readFile(
+        `${this.myPath}/${this.name}`,
+        'utf8'
+      );
+      this.store = JSON.parse(setting);
+      const root = `${os.homedir()}/${AppConfig.DEFAULT_WORKSPACE_NAME}`;
+      await new WorkspaceMigration(userSettingService.get().VERSION, root).up();
+      this.isValidConfig(this.store);
+    } catch(error:any) {
+      log.error("[ WORKSPACE CONFIG ]:", "Invalid workspace configuration");
+      const ws =  await fs.promises.readFile(
+        `${this.myPath}/${this.name}`,
+        'utf8'
+      );
+      await fs.promises.writeFile(`${this.myPath}/workspaceCfg-invalid.json`,ws);
+      this.store = this.defaultStore;
+    }
   }
 
+private isValidConfig(config: Partial<IWorkspaceCfg>) { // Keep updated with workspace migrations
```

Comment:
@agustingroh a little question here. Each time we add a new para in the config, should we rewrite this function?